app/models/private_message_topic_tracking_state.rb

```diff
@@ -123,6 +124,12 @@ def self.publish_unread(post)
       .select([:user_id, :last_read_post_number, :notification_level])
       .each do |tu|
 
+      if tu.last_read_post_number.nil? &&
+          topic.created_at < tu.user.user_option.treat_as_new_topic_start_date
```

Comment:
can this be `nil` ? is it guaranteed to be correct for all users ?

Reply:
@SamSaffron Yup it'll be `nil` if you're part of a group that has access to the message.

Reply:
Nope it'll always return a datetime object per 

https://github.com/discourse/discourse/blob/9998090e5b045d3acedf09820585eef198b915b4/app/models/user_option.rb#L145-L161